### ISSUE TITLE
/api/v2/silences: context deadline exceeded

### ISSUE THREAD

**Issue by itatabitovski** [prymitive/karma#791]
After upgrading from Alertmanager 0.15.3 to .0.17.0, Karma 0.39 stopped working.

curling the endpoint works without any issues.

```
./karma-linux-amd64 --version
"v0.39"
```

```
alertmanager --version
alertmanager, version 0.17.0 (branch: non-git, revision: c7551cd)
  build date:       20190624-21:36:45
  go version:       go1.11.10
```

```
INFO[0000] [am] Configured Alertmanager source at https://alertmanager:9093 (proxied: true) 
INFO[0000] Initial Alertmanager query                   
INFO[0000] Pulling latest alerts and silences from Alertmanager 
INFO[0000] [am] Collecting alerts and silences    
INFO[0000] GET https://alertmanager:9093/metrics timeout=0s 
INFO[0000] Upstream version: 0.17.0                     
INFO[0000] GET https://alertmanager:9093/api/v1/status timeout=0s 
INFO[0000] [am] Remote Alertmanager version: 0.17.0 
ERRO[0000] [am] Get https://alertmanager:9093/api/v2/silences: context deadline exceeded```

**Comment by prymitive** [comment]
Well, you have 0s timeout for all requests: `timeout=0s`

**Comment by itatabitovski** [comment]
@prymitive thanks, I didn't remember to check the docs because there were no issues with 0.15.3... Would it make sense to have some default timeout instead of 0s?

**Comment by prymitive** [comment]
I though it was defaulting to 40s but looks like that's not true, I removed timeout from my config and I got `0s`. This definetely needs fixing, thanks for pointing that out

**Comment by prymitive** [comment]
Will be fixed via #792

**Comment by prymitive** [comment]
Should be fixed in master
